wagtail/admin/rich_text/converters/html_to_contentstate.py

```diff
@@ -206,12 +206,19 @@ def get_attribute_data(self, attrs):
         try:
             page = Page.objects.get(id=attrs['id']).specific
         except Page.DoesNotExist:
-            return {}
+            # retain ID so that it's still identified as a page link (albeit a broken one)
+            return {
+                'id': int(attrs['id']),
+                'url': None,
+                'parentId': None
+            }
```

Comment:
From what I understand, this combination of having an id and not having a URL will only happen for broken page links. Could we leverage this to display them differently in the editor? e.g. use some kind of "warning" UI cues, and potentially have a label that says "this links look broken".

This doesn't need to be done as part of this PR of course, but it seems like a nice potential improvement.

Edit: here, https://github.com/wagtail/wagtail/blob/09cddd81f69660e5d8836eb205355c917dc9b49b/client/src/components/Draftail/decorators/Link.js#L20-L22

and then there,

https://github.com/wagtail/wagtail/blob/09cddd81f69660e5d8836eb205355c917dc9b49b/client/src/components/Draftail/decorators/TooltipEntity.js#L105-L113

we use the URL as a label for the link tooltip. I will have a look at what happens if this is set to `None`.